### DISCUSSION TITLE
Troubles with calibrating lighthouse base stations

### DISCUSSION THREAD

**Discussion by CRhodes-123** [bitcraze/discussions#278] [Q&A]
Hello, I am a Mechanical Engineering undergraduate student at the University of Houston, and I do research with crazyflies. Right now I am trying to set up the lighthouse base stations, and I have been having some troubles obtaining the calibration data. 

For reference, I am running everything through the virtual machine (The virtual machine is the BitcrazeVM and I am running it through Oracle VM on a Dell laptop). We are setting up the V2 lighthouse base stations, and we have two of them. They are already mounted correctly and plugged in.

I have found the get_lh2_calib_data.py file, but when I run it through the terminal I am having trouble figuring out what the serial port is supposed to be. This is described in the code as "usually something like /dev/tty.usb123456" under lines 34 & 35. Under the dev folders I can find the tty files, but they are all named as tty, tty1, tty2 and so forth (they go to tty63, and there are also some named as ttyS1, ttyS2 and so forth to ttyS31). 

When I am running the file in the terminal, this is what I am running:
~/Desktop/projects/crazyflie-firmware/tools/lighthouse$ /get_lh2_calib_data.py --dev /dev/tty.usb123456

When I run the code omitting the .usb123456 at the end, it says there is no such file or directory, and when I run the code through the visual studio code and click the button to run python file in terminal, when I omit the .usb123456 at the end I get errors about not having the correct index, and this has been the case for every tty file that I have tried.

I hope I have provided enough information for you to help me out with this, and if I have left anything important out please just let me know!

Thanks

**Comment by knmcguire** [comment]
Hi! Please use the tutorial here: https://www.bitcraze.io/documentation/tutorials/getting-started-with-lighthouse/. You should be able to all of this from the CFclient which is the official way that we propose.

If you are having trouble to connect to the basestations in the VM, these instructions for works in windows 10/11 as well. Please follow the instructions for installation in windows: https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/installation/install/
